src/Service.Tests/Unittests/SerializationDeserializationTests.cs

```diff
@@ -305,6 +369,45 @@ private void InitializeObjects()
             _databaseStoredProcedure.StoredProcedureDefinition.Parameters.Add("Id", _parameterDefinition);
         }
 
+        /// <summary>
+        /// During serialization we add TypeName to the DatabaseObject based on its type.
+        /// If the code refactor then this change will catch that as GraphQL workload would be impacted.
+        /// We will have artifacts with TypeName with old project directory/ location of this class, hence during deserialization if object is not in
+        /// that location it will fail.
+        /// [Note] : is this test fails notify the graphql workload team.
+        /// </summary>
+        private void TestTypeNameChanges(DatabaseObject databaseobject, string objectName)
+        {
+            Dictionary<string, DatabaseObject> dict = new();
+            dict.Add("person", databaseobject);
+            string jsonString = JsonSerializer.Serialize(dict, _options);
+
+            // Find the start index of TypeName
+            int typeNameIndex = jsonString.IndexOf("\"TypeName\":\"", StringComparison.Ordinal);
+
+            if (typeNameIndex != -1)
```

Comment:
what happens when `typeNameIndex` is 1? should that result in a failure?

Reply:
good catch, yes, if TypeNameIndex == -1 means TypeName substring not found in the serialized string, which is incorrect, I have added the failure else case - will just fail the test.